starport_template/lib/widgets/app_version_text.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:cosmos_utils/app_info_extractor.dart';
+import 'package:cosmos_utils/app_info_provider.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+/// TODO: Make this intelligent enough to display all of the properties inside [AppInfo] configurable from outside
+
+class AppVersionText extends StatefulWidget {
```

Comment:
you can make this widget a `StatelessWidget` and wrap the `appInfoProvider.getAppVersion()` into a FutureBuilder now. something like:

```dart
FutureBuilder<String>(
        future: appInfoProvider.getAppVersion(),
        builder: (context, snapshot) => Text(snapshot.data ?? ""),
      );
```